doc/design/diagrams/request_response/client_state_machine.puml

```diff
@@ -0,0 +1,22 @@
+@startuml
+
+skinparam backgroundColor #EEEBDC
+
+[*] -down-> NOT_CONNECTED
+
+NOT_CONNECTED -left-> NOT_CONNECTED : OFFER
+NOT_CONNECTED -right-> CONNECT_REQUESTED : CONNECT
```

Comment:
I'm not quite sure if the `CONNECT_REQUESTED` state should be renamed to `WAIT_FOR_OFFER` or `WAIT_FOR_SERVER`